test/kibana_test_server_url_parts.js

```diff
@@ -1,10 +1,38 @@
 import { kibanaUser } from './shield';
+import url from 'url';
+
+let testKibanaUrl;
+let testKibanaProtocol;
+let testKibanaHostname;
+let testKibanaPort;
+let testKibanaUsername;
+let testKibanaPassword;
+
+// Allow setting one complete TEST_KIBANA_URL for Kibana like https://elastic:changeme@myCloudInstance:5601
+if (process.env.TEST_KIBANA_URL) {
+  testKibanaUrl = url.parse(process.env.TEST_KIBANA_URL);
+  // have to remove the ":" off protocol
+  testKibanaProtocol = testKibanaUrl.protocol.slice(0, -1);
+  testKibanaHostname = testKibanaUrl.hostname;
+  testKibanaPort = parseInt(testKibanaUrl.port, 10);
+  testKibanaUsername = testKibanaUrl.username;
+  testKibanaPassword = testKibanaUrl.password;
+} else {
+  // Allow setting any individual component(s) of the URL,
+  // or use default values (username and password from shield.js)
+  testKibanaProtocol = process.env.TEST_KIBANA_PROTOCOL || 'http';
+  testKibanaHostname = process.env.TEST_KIBANA_HOSTNAME || 'localhost';
+  testKibanaPort = parseInt(process.env.TEST_KIBANA_PORT, 10) || 5620;
+  testKibanaUsername = process.env.TEST_KIBANA_USERNAME || kibanaUser.username;
+  testKibanaPassword = process.env.TEST_KIBANA_PASSWORD || kibanaUser.password;
+}
 
 export const kibanaTestServerUrlParts = {
-  protocol: process.env.TEST_KIBANA_PROTOCOL || 'http',
-  hostname: process.env.TEST_KIBANA_HOSTNAME || 'localhost',
-  port: parseInt(process.env.TEST_KIBANA_PORT, 10) || 5620,
-  auth: kibanaUser.username + ':' + kibanaUser.password,
-  username: kibanaUser.username,
-  password: kibanaUser.password,
+
+  protocol: testKibanaProtocol,
+  hostname: testKibanaHostname,
+  port: testKibanaPort,
+  username: testKibanaUsername,
+  password: testKibanaPassword,
+  auth: `${testKibanaUsername}:${testKibanaPassword}`
```

Comment:
```js
import { kibanaUser } from './shield';
import url from 'url';

let protocol;
let hostname;
let port;
let username;
let password;
let auth;

// Allow setting one complete TEST_KIBANA_URL for Kibana like
// https://elastic:changeme@myCloudInstance:5601
if (process.env.TEST_KIBANA_URL) {
  const testKibanaUrl = url.parse(process.env.TEST_KIBANA_URL);
  // have to remove the ":" off protocol
  protocol = testKibanaUrl.protocol.slice(0, -1);
  hostname = testKibanaUrl.hostname;
  port = parseInt(testKibanaUrl.port, 10);
  username = testKibanaUrl.username;
  password = testKibanaUrl.password;
} else {
  // Allow setting any individual component(s) of the URL,
  // or use default values (username and password from shield.js)
  protocol = process.env.TEST_KIBANA_PROTOCOL || 'http';
  hostname = process.env.TEST_KIBANA_HOSTNAME || 'localhost';
  port = parseInt(process.env.TEST_KIBANA_PORT, 10) || 5620;
  username = process.env.TEST_KIBANA_USERNAME || kibanaUser.username;
  password = process.env.TEST_KIBANA_PASSWORD || kibanaUser.password;
}

export const kibanaTestServerUrlParts = {
  protocol,
  hostname,
  port,
  auth: `${username}:${password}`,
  username,
  password,
};
```
maybe make things more consistent with the ES file, too.

Reply:
Thanks for the suggestion!  I made that change and it is cleaner.